src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -19,6 +19,34 @@
 #include <Kokkos_Sort.hpp> // min_max_functor
 #include <Kokkos_View.hpp>
 
+#if defined(KOKKOS_ENABLE_CUDA)
+#if (KOKKOS_COMPILER_CLANG < 900)
+// Clang of version less than 9.0 cannot compile Thrust, failing with errors
+// like this:
+//    <snip>/thrust/system/cuda/detail/core/agent_launcher.h:557:11:
+//    error: use of undeclared identifier 'va_printf'
+// Defining _CubLog here allows us to avoid that code path, however disabling
+// some debugging diagnostics
+//
+// If _CubLog is already defined, we save it into _CubLog_save, and restore at
+// the end
+#ifdef _CubLog
+#define _CubLog_save _CubLog
```

Comment:
Can you use something like `ARBORX_Cublog_save` instead of  `_Cublog_save`  (`identifiers that begin with an underscore followed by an uppercase letter are reserved` from [here](https://en.cppreference.com/w/cpp/language/identifiers)) Another way to do this logic is: 
```
#if (KOKKOS_COMPILER_CLANG < 900) && !defined(_Cublog)
  #define _CubLog
  #include <thrust/device_ptr.h>
  #include <thrust/sort.h>
  #undef _CubLog
#else 
  #include <thrust/device_ptr.h>
  #include <thrust/sort.h>
#endif

Reply:
Why? I undef the macro only if it was not defined before

Reply:
Andrey reported it needs to be defined empty.

Reply:
I *think* I tried without defining it, but not 100% sure anymore. Let me try @Rombur's patch and see if CI still passes.

Reply:
`_CubLog` has to be defined to something that can be compiled prior to including Thrust headers. Defining it to empty is the easiest thing. It does not matter whether it was defined prior or not, we cannot guarantee anything unless we define it ourselves.